cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -811,6 +813,47 @@ def _validate_sparsity(self):
                         f"to use this type of matrix for the given sparsity."
                     )
 
+    def _validate_seurat_convertibility(self):
+        """
+        Use length of component matrices to determine if the anndata object will be unable to be converted to Seurat by
+        virtue of the R language's array size limit (4-byte signed int length). Add warning for each matrix which is
+        too large.
+        rtype: None
+        """
+        to_validate = [self.adata.X]
+        to_validate_name = ["X"]
+        # check if there's raw data
+        if self.adata.raw:
+            to_validate.append(self.adata.raw.X)
+            to_validate_name.append("raw")
+        failing_matrix_names: List[str] = []
+        # Check length of component arrays
+        for x, x_name in zip(to_validate, to_validate_name):
+            if (
+                    isinstance(x, sparse.csc_matrix)
+                    or isinstance(x, sparse.csr_matrix)
+                    or isinstance(x, sparse.coo_matrix)
+            ):
+                effective_r_array_size = x.count_nonzero()
+            elif isinstance(x, ndarray):
+                effective_r_array_size = max(x.shape)
+            else:
+                self.warnings.append(
+                    f"{x_name} matrix is of type {type(x)}, sparsity calculation hasn't been "
```

Comment:
Is this message been defined somewhere? I don't believe this is readable to users.

Reply:
The warnings get displayed: 

https://github.com/chanzuckerberg/single-cell-curation/blob/6e598e8f1d92532c8f59ede08668aa8a3d9d2194/cellxgene_schema_cli/cellxgene_schema/validate.py#L1277

Reply:
Ah, sorry, ignore that. I need to remove that chunk. Technically we should be warning whenever the matrix is not of the two previous types (scipy.sparse, ndarray), but I left this in after some copy/paste. Will remove.